transcendence-app/src/chat/infrastructure/db/postgres/chatPostgres.repository.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@nestjs/common';
+import { BasePostgresRepository } from '../../../../shared/db/postgres/postgres.repository';
+import { table } from '../../../../shared/db/models';
+import { PostgresPool } from '../../../../shared/db/postgres/postgresConnection.provider';
+import { ChatsPaginationQueryDto } from '../../../dto/chat.pagination.dto';
+import {
+  entityQueryMapper,
+  makeQuery,
+} from '../../../../shared/db/postgres/utils';
+import { BooleanString } from '../../../../shared/enums/boolean-string.enum';
+import { LocalFileEntity } from '../../../../shared/local-file/infrastructure/db/local-file.entity';
+import { PoolClient } from 'pg';
+import { ChatEntity, chatKeys } from '../chat.entity';
+import { IChatRepository } from '../chat.repository';
+import { UpdateChatDto } from '../../../../chat/dto/update-chat.dto';
+
+@Injectable()
+export class ChatPostgresRepository
+  extends BasePostgresRepository<ChatEntity>
+  implements IChatRepository
+{
+  constructor(protected pool: PostgresPool) {
+    super(pool, table.CHATS);
+  }
+  async getById(id: string): Promise<ChatEntity | null> {
+    const chats = await this.getByKey(chatKeys.ID, id);
+    return chats && chats.length ? chats[0] : null;
+  }
+
+  async getByChatName(chatName: string): Promise<ChatEntity | null> {
+    const chats = await this.getByKey(chatKeys.NAME, chatName);
+    return chats && chats.length ? chats[0] : null;
+  }
+
+  async deleteByChatName(chatName: string): Promise<ChatEntity | null> {
+    const chats = await this.deleteByKey(chatKeys.NAME, chatName);
+    return chats && chats.length ? chats[0] : null;
+  }
+
+  async updateById(
+    id: string,
+    updateChatDto: UpdateChatDto,
+  ): Promise<ChatEntity | null> {
+    const chats = await this.updateByKey(chatKeys.ID, id, updateChatDto);
+    return chats && chats.length ? chats[0] : null;
+  }
+
+  getPaginatedChats(
+    paginationDto: Required<ChatsPaginationQueryDto>,
+  ): Promise<ChatEntity[] | null> {
+    const { limit, offset, sort, search } = paginationDto;
+    const orderBy = sort === BooleanString.True ? chatKeys.NAME : chatKeys.ID;
+    return makeQuery<ChatEntity>(this.pool, {
+      text: `SELECT *
+      FROM ${this.table}
+      WHERE ${chatKeys.NAME} ILIKE $1
+      ORDER BY ${orderBy}
+      LIMIT $2
+      OFFSET $3;`,
+      values: [`%${search}%`, limit, offset],
+    });
+  }
```

Comment:
not yours, but I wanted to point this out. Since this query is being used in many places now, we may move it to the generic base repository. That is the idea, DRY principle! (don't repeat yourself :smile)

cc @gbudau @Damrod

Reply:
We could create a new task for this to not block this PR :+1: